go/libraries/doltcore/sqle/dprocedures/dolt_remote.go

```diff
@@ -46,6 +47,9 @@ func doDoltRemote(ctx *sql.Context, args []string) (int, error) {
 	if len(dbName) == 0 {
 		return 1, fmt.Errorf("Empty database name.")
 	}
+	if err := branch_control.CheckAccess(ctx, branch_control.Permissions_Write); err != nil {
```

Comment:
This is overly broad, the remote command doesn't actually change the branch head in most cases

Reply:
But it does modify branch information, hence the guard. You don't want someone to just add or remove remotes without permission.

Reply:
Remotes are separate from branches. What am I missing?

Reply:
Oh I’m thinking of tracking remotes, not the remotes themselves. My bad.